dm/master/scheduler/scheduler.go

```diff
@@ -384,6 +405,111 @@ func (s *Scheduler) GetSourceCfgByID(source string) *config.SourceConfig {
 	return &clone
 }
 
+// transferWorkerAndSource transfer worker and source bound relations.
+// lworker, "", "", rsource				This means a unbounded source bounded to a free worker
+// lworker, lsource, rworker, "" 		This means transfer a source from a worker to another free worker
+// lworker, lsource, "", rsource		This means transfer a worker from a bounded source to another unbounded source
+// lworker, lsource, rworker, rsource	This means transfer two bounded relations.
+func (s *Scheduler) transferWorkerAndSource(lworker, lsource, rworker, rsource string) error {
```

Comment:
i think there is too much magic here, can we split this func to four independent func ？

just a advice (not strong , wait for another reviewer comment

Reply:
I think they have the same logic: unbound; new bound; try bound for free worker/unbound source, so keep them in to one func.